pipelines/tekton/aiedge-e2e/aiedge-e2e.pipeline.yaml

```diff
@@ -116,24 +124,54 @@ spec:
     description: The buildah version used to build the model container image
     value: $(tasks.retrieve-build-image-info.results.buildah-version)
   tasks:
+  - name: fetch-model-registry
```

Comment:
Shouldn't we split into 2 tasks for better clarity? This is both fetching model registry + it also processes data.

Probably it would be better to make this task optional (just run in case the model registry is enabled) and then the subsequent task would either override data from the fetch model registry task or use the pipeline as the source. I get that this would mean to have double the amount of params as one set will be for pipelineRun and the second set would be for parameter values coming from the fetch model registry task. But I would still find it more readable.

Another option would be to name this task better, something like "fetch-model-location-params" or something like that. Because this is what it does, fetching the values either from the registry or from the pipeline run.

Reply:
> Shouldn't we split into 2 tasks for better clarity? This is both fetching model registry + it also processes data.
So IMO it's just fetch the data from the model registry so I don't know how to break this into 2 tasks. There is no connection steps required there is just curl commands for the data. 

I like the override data part for the pipeline, but I don't think that would be syntactically allowed by Tekton. I can't think of way to configure tekton pipeline as to pass params.xyz when model-registry enabled OR task.results.xyz when model-registry enabled.  @grdryn  WDYT?  

We could rename the task but I wanted name to specify that it's involving MR. @MarianMacik  Does that make sense?

Reply:
@biswassri For that override, we would need to have 2 set of input params for the task which would decide which data to use (either from the model registry or from the pipeline). So if the first task was run (which fetched the model registry), the set of parameters coming from that task's output would be filled in. Then the "decision" task would check if these params are set and use that one as its output or it would just take the values from the pipeline run input params and set it as the output of the task. But as I said, this would mean to have input parameters of that "decision" task doubled as one set would be coming from the model registry task (in case it was not run, it would be empty) and the other set would be coming from the pipeline run params. And the "decision" task would decide which values to put as the output (which will be then used as input for the s3/git task), either model registry values or the pipeline run values.


But this puts unnecessary overhead, I think. So just a renaming might be better. `fetch-model-registry-or-pipelinerun-params` or something similar, maybe?

Reply:
@MarianMacik My intention with that with this task and that hopefully we would eventually move out of the params from pipelinerun and only use MR for params. So in that case this would be the startpoint of the pipeline always. 

I was thinking of naming this fetch-model-params eventually we can implement ^what you are suggesting and this would be that decision task. WDYT?

Reply:
@biswassri Agree. `fetch-model-params` sounds better.